soc/arm/rpi_pico/rp2/binary_info.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <soc.h>
+
+#include <pico/binary_info.h>
+#include <boot_stage2/config.h>
+
+#include <version.h>
+
+#if CONFIG_RP2_BINARY_INFO_PROGRAM_NAME_OVERRIDE
+#undef PICO_PROGRAM_NAME
+#define PICO_PROGRAM_NAME CONFIG_RP2_BINARY_INFO_PROGRAM_NAME
+#else
+#ifndef PICO_PROGRAM_NAME
+#define PICO_PROGRAM_NAME APPLICATION_SOURCE_DIR_BASE
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_BOARD_OVERRIDE
+#undef PICO_BOARD
+#define PICO_BOARD CONFIG_RP2_BINARY_INFO_BOARD
+#else
+#ifndef PICO_BOARD
+#define PICO_BOARD CONFIG_BOARD
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_SDK_VERSION_STRING_OVERRIDE
+#undef PICO_SDK_VERSION_STRING
+#define PICO_SDK_VERSION_STRING CONFIG_RP2_BINARY_INFO_SDK_VERSION_STRING
+#else
+#ifndef PICO_SDK_VERSION_STRING
+#define PICO_SDK_VERSION_STRING RPI_PICO_SDK_VERSION_STRING
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_PROGRAM_VERSION_STRING_OVERRIDE
+#undef PICO_PROGRAM_VERSION_STRING
+#define PICO_PROGRAM_VERSION_STRING CONFIG_RP2_BINARY_INFO_PROGRAM_VERSION_STRING
+#else
+#ifndef PICO_PROGRAM_VERSION_STRING
+#define PICO_PROGRAM_VERSION_STRING STRINGIFY(BUILD_VERSION)
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_PROGRAM_BUILD_DATE_OVERRIDE
+#undef PICO_PROGRAM_BUILD_DATE
+#define PICO_PROGRAM_BUILD_DATE CONFIG_RP2_BINARY_INFO_PROGRAM_BUILD_DATE
+#ifndef PICO_PROGRAM_BUILD_DATE
+#define PICO_PROGRAM_BUILD_DATE __DATE__
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_PROGRAM_DESCRIPTION_OVERRIDE
+#undef PICO_PROGRAM_DESCRIPTION
+#define PICO_PROGRAM_DESCRIPTION CONFIG_RP2_BINARY_INFO_PROGRAM_DESCRIPTION
+#else
+#ifndef PICO_PROGRAM_DESCRIPTION
+#define PICO_PROGRAM_DESCRIPTION CONFIG_RP2_BINARY_INFO_PROGRAM_DESCRIPTION
+#endif
+#endif
+
+#if CONFIG_RP2_BINARY_INFO_PROGRAM_URL_OVERRIDE
+#undef PICO_PROGRAM_URL
+#define PICO_PROGRAM_URL CONFIG_RP2_BINARY_INFO_PROGRAM_URL
+#else
+#ifndef PICO_PROGRAM_URL
+#define PICO_PROGRAM_URL CONFIG_RP2_BINARY_INFO_PROGRAM_URL
+#endif
+#endif
+
+extern uint32_t __rom_region_end;
+bi_decl(bi_binary_end((intptr_t)&__rom_region_end));
+
+#ifdef PICO_PROGRAM_NAME
+bi_decl(bi_program_name((uint32_t)PICO_PROGRAM_NAME));
+#endif
+
+#ifdef PICO_BOARD
```

Comment:
There are no strong reason to configure the settings by CMake variables. Removed it.
(It is port from Pico-SDK. It may sometimes convenient, but no strong reason to keep it.)

Reply:
Thanks, resolving